buildpack/dockerfile.go

```diff
@@ -64,80 +82,113 @@ func VerifyBuildDockerfile(dockerfile string, logger log.Logger) error {
 
 	// validate only 1 FROM
 	if len(stages) > 1 {
-		return fmt.Errorf("build.Dockerfile is not permitted to use multistage build")
+		return fmt.Errorf(errMultiStageNotPermitted, buildDockerfileName)
 	}
 
 	// validate only permitted Commands
 	for _, stage := range stages {
 		for _, command := range stage.Commands {
 			found := false
-			for _, permittedCommand := range permittedCommandsBuild {
+			for _, permittedCommand := range permittedCommands {
 				if permittedCommand == strings.ToUpper(command.Name()) {
 					found = true
 					break
 				}
 			}
 			if !found {
-				logger.Warnf("build.Dockerfile command %s on line %d is not recommended", strings.ToUpper(command.Name()), command.Location()[0].Start.Line)
+				logger.Warnf(warnCommandNotRecommended, buildDockerfileName, strings.ToUpper(command.Name()), command.Location()[0].Start.Line)
 			}
 		}
 	}
 
 	// validate build.Dockerfile preamble
 	if len(margs) != 1 {
-		return fmt.Errorf("build.Dockerfile did not start with required ARG command")
+		return errors.New(errBuildMissingRequiredARGCommand)
 	}
-	if margs[0].Args[0].Key != "base_image" {
-		return fmt.Errorf("build.Dockerfile did not start with required ARG base_image command")
+	if margs[0].Args[0].Key != baseImageArgName {
+		return errors.New(errBuildMissingRequiredARGCommand)
 	}
 	// sanity check to prevent panic
 	if len(stages) == 0 {
-		return fmt.Errorf("build.Dockerfile should have at least one stage")
+		return fmt.Errorf(errMissingRequiredStage, buildDockerfileName)
 	}
-	if stages[0].BaseName != "${base_image}" {
-		return fmt.Errorf("build.Dockerfile did not contain required FROM ${base_image} command")
+
+	if stages[0].BaseName != baseImageArgRef {
+		return errors.New(errBuildMissingRequiredFROMCommand)
 	}
 
 	return nil
 }
 
-func VerifyRunDockerfile(dockerfile string) error {
+func VerifyRunDockerfile(dockerfile string, buildpackAPI *api.Version, logger log.Logger) (string, error) {
+	if buildpackAPI.LessThan("0.10") {
```

Comment:
I'm surprised we are keeping this branching logic - do we have such rules for experimental features?

Reply:
Yeah... maybe it's a bit overzealous. I can remove it

Reply:
Updated in 34e396a3